draft-ietf-teep-protocol.xml

```diff
@@ -227,32 +227,91 @@ otherwise.
 
     <section anchor="detailmsg" title="Detailed Messages Specification">
 
-    <t>The CBOR-encoded messages are protected by COSE, as described in CDDL format <xref target="I-D.ietf-cbor-cddl"/> below. 
+    <t>TEEP messages are protected by the COSE_Sign1 structure.
+      The TEEP protocol messages are described in CDDL format <xref target="I-D.ietf-cbor-cddl"/> below. 
     
  <figure>
  <artwork><![CDATA[
-Outer_Wrapper = {
-    msg-authenc-wrapper         => bstr .cbor
-                                   Msg_AuthEnc_Wrapper / nil,
     teep-message                => (QueryRequest / 
                                     QueryResponse / 
                                     TrustedAppInstall / 
                                     TrustedAppDelete / 
                                     Error / 
                                     Success ), 
 }
-
-msg-authenc-wrapper = 1
-teep-message = 2
-
-Msg_AuthEnc_Wrapper = [ * (COSE_Mac_Tagged / 
-                           COSE_Sign_Tagged / 
-                           COSE_Mac0_Tagged / 
-                           COSE_Sign1_Tagged)]
                        ]]></artwork>
           </figure>
      </t> 
 
+ <section title="Creating and Validating Teep Messages">
+
+<section title="Creating a Teep message">
+
+  <t> To create a Teep message, the following steps are performed.</t>
+
+<t>        
+<list style="numbers"> 
+
+   <t> Create a Teep message according to the description below and populate 
+       it with the respective content.</t>
+
+   <t>  Create a COSE Header containing the desired set of Header
+       Parameters.  The COSE Header MUST be valid per the [RFC8152]
+       specification.</t>
+
```

Comment:
Need to explain here where encryption comes in... messages should be encrypted first when confidentiality is needed. and THEN signed.  (OTrP already worked that way.)

Reply:
This needs to be described in a dedicated paragraph